### PR TITLE
Add a dev stage

### PR DESCRIPTION
**Pull request by ttung**
1. Dev points to a separate bucket.
2. tests continue running against org-humancellatlas-dss-test

### CODE REVIEW

File: chalice/app.py [hunk 1]
```diff
@@ -7,10 +7,14 @@
 pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), 'chalicelib')) # noqa
 sys.path.insert(0, pkg_root) # noqa
 
-from dss import create_app
+from dss import Config, create_app
 from dss.events.handlers.sync import sync_blob
 from dss.util import paginate
 
+
+Config.set_config_by_env()
```
Comment (kislyuk) [inline chalice/app.py]: Should this happen implicitly when importing dss?

File: dss/__init__.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 from connexion.resolver import RestyResolver
 from flask_failsafe import failsafe
 
+from .config import BucketConfig, Config
```
Comment (kislyuk) [inline dss/__init__.py]: Could we name BucketConfig something more differentiated? BucketStage maybe?

File: dss/config.py [hunk 3]
```diff
@@ -12,15 +12,32 @@
 
 
 class BucketConfig(Enum):
-    PROD = 0
-    TEST = 1
-    TEST_FIXTURE = 2
+    ILLEGAL = auto()
```
Comment (kislyuk) [inline dss/config.py]: 🚔

File: chalice/app.py (hunk 1 above)
Comment (ttung) [inline chalice/app.py]: no, because tests would then start writing to the dev bucket.

File: dss/__init__.py (hunk 2 above)
Comment (ttung) [inline dss/__init__.py]: Sure.

**Pull request merged**